lib/sisimai/bite/email/mxlogic.rb

```diff
@@ -0,0 +1,253 @@
+        |Delay[ ]reason:[ ]
+        |Message[ ].+[ ](?:has[ ]been[ ]frozen|was[ ]frozen[ ]on[ ]arrival[ ]by[ ])
+        )
+      }x
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'McAfee SaaS'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      # X-MX-Bounce: mta/src/queue/bounce
+      # X-MXL-NoteHash: ffffffffffffffff-0000000000000000000000000000000000000000
+      # X-MXL-Hash: 4c9d4d411993da17-bbd4212b6c887f6c23bab7db4bd87ef5edc00758
+      def headerlist;  return ['X-MXL-NoteHash', 'X-MXL-Hash', 'X-MX-Bounce']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from MXLogic
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['x-mx-bounce']
+        match += 1 if mhead['x-mxl-hash']
+        match += 1 if mhead['x-mxl-notehash']
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        match += 1 if mhead['from']    =~ Re0[:from]
+        return nil if match.zero?
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        localhost0 = ''     # (String) Local MTA
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:'message-rfc822']).zero?` instead of `readcursor & Indicators[:'message-rfc822'] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->